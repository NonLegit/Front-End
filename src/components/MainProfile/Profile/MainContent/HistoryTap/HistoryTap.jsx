import { useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import { WideBox } from '../styles';
import EmptyContent from '../../../EmptyContent/EmptyContent';
import Post from '../Posts/Post';
/**
 * HistoryTap
 *
 * @component HistoryTap
 * @returns {React.Component} HistoryTap
 */
function HistoryTap() {
  const { subTitle } = useParams();
  const [isContent, setIsContent] = useState(false);
  const [content, setContent] = useState([]);

  useEffect(() => {
    setContent(JSON.parse(localStorage.getItem('RedditHistory')));
    if (content?.length > 0) { setIsContent(true); }
  }, [content, subTitle]);

  const emptyContent = 'hmm... looks like you haven\'t recently viewed anything yet';
  return (
    <WideBox>
      {!isContent && <EmptyContent emptyContent={emptyContent} />}
      {isContent && content.map((entity, index) => (<Post key={`${index + 0}`} entity={entity} />))}
    </WideBox>
  );
}

export default HistoryTap;
