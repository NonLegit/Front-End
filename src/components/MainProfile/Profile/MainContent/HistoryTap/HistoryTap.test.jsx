import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import HistoryTap from './HistoryTap';

test('test snapshot', async () => {
  const tree = renderer.create(
    <HistoryTap />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
