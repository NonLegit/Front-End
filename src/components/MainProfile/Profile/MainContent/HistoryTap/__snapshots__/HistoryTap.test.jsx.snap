// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  className="MuiBox-root css-qsaw8"
>
  <div
    className="MuiBox-root css-1bk1a37"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-reyiuj-MuiSvgIcon-root"
      data-testid="ForwardOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
      />
    </svg>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-e5mdo1-MuiSvgIcon-root"
      data-testid="ForwardOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
      />
    </svg>
  </div>
  <div
    className="MuiBox-root css-1bk1a37"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-reyiuj-MuiSvgIcon-root"
      data-testid="ForwardOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
      />
    </svg>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-e5mdo1-MuiSvgIcon-root"
      data-testid="ForwardOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
      />
    </svg>
  </div>
  <div
    className="MuiBox-root css-1bk1a37"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-reyiuj-MuiSvgIcon-root"
      data-testid="ForwardOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
      />
    </svg>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-e5mdo1-MuiSvgIcon-root"
      data-testid="ForwardOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
      />
    </svg>
  </div>
  <div
    className="MuiBox-root css-1goo8qo"
  >
    <div
      className="MuiBox-root css-5pmy0u"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-reyiuj-MuiSvgIcon-root"
        data-testid="ForwardOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
        />
      </svg>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-e5mdo1-MuiSvgIcon-root"
        data-testid="ForwardOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
        />
      </svg>
    </div>
    <p
      className="MuiTypography-root MuiTypography-body1 css-c6mk4m-MuiTypography-root"
    >
      hmm... looks like you haven't recently viewed anything yet
    </p>
  </div>
  <div
    className="MuiBox-root css-1bk1a37"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-reyiuj-MuiSvgIcon-root"
      data-testid="ForwardOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
      />
    </svg>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-e5mdo1-MuiSvgIcon-root"
      data-testid="ForwardOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
      />
    </svg>
  </div>
  <div
    className="MuiBox-root css-1bk1a37"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-reyiuj-MuiSvgIcon-root"
      data-testid="ForwardOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
      />
    </svg>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-e5mdo1-MuiSvgIcon-root"
      data-testid="ForwardOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
      />
    </svg>
  </div>
  <div
    className="MuiBox-root css-1bk1a37"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-reyiuj-MuiSvgIcon-root"
      data-testid="ForwardOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
      />
    </svg>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-e5mdo1-MuiSvgIcon-root"
      data-testid="ForwardOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
      />
    </svg>
  </div>
</div>
`;
